frontend/packages/knative-plugin/src/hooks/useEventSourceStatus.ts

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { checkAccess } from '@console/internal/components/utils';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { getEventSourceMetadata, getKameletMetadata } from '../utils/create-eventsources-utils';
+import { EventSourceMetaData } from '../components/add/import-types';
+import { CamelKameletBindingModel, CamelKameletModel } from '../models';
+import {
+  isDynamicEventSourceKind,
+  useEventSourceModels,
+} from '../utils/fetch-dynamic-eventsources-utils';
+
+export const useEventSourceStatus = (
+  namespace: string,
+  sourceKindProp?: string,
+  kameletName?: string,
+): {
+  sourceFound: boolean;
+  createSourceAccess: boolean;
+  loaded: boolean;
+  normalizedSource: EventSourceMetaData;
+  kamelet: K8sResourceKind;
+} => {
+  const { t } = useTranslation();
+  const { eventSourceModels, loaded: eventSourceModelsLoaded } = useEventSourceModels();
+  const [kamelet, kameletLoaded] = useK8sGet<K8sResourceKind>(
+    CamelKameletModel,
+    kameletName,
+    namespace,
+  );
+  const [createSourceAccess, setCreateSourceAccess] = React.useState<boolean>(true);
+  const [sourceFound, setSourceFound] = React.useState<boolean>(true);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [normalizedSource, setNormalizedSource] = React.useState<EventSourceMetaData>();
+
+  const isEventSource = sourceKindProp && isDynamicEventSourceKind(sourceKindProp);
+  const isKameletSource = kameletName && sourceKindProp === CamelKameletBindingModel.kind;
+  const isSourceKindPresent = sourceKindProp || isKameletSource;
+
+  const eventSourceModel =
+    isEventSource && eventSourceModels?.find((model: K8sKind) => model.kind === sourceKindProp);
+  const sourceModel =
+    isKameletSource && kamelet && kameletLoaded ? CamelKameletBindingModel : eventSourceModel;
+
+  React.useEffect(() => {
+    let sourceData = {};
+    if (sourceModel) {
+      setSourceFound(true);
+      checkAccess({
```

Comment:
we can use `useAccessReview` instead of this.

Reply:
yeah made use of `useAccessReview2` as needed loading state